libmamba/src/solv-cpp/solvable.hpp

```diff
@@ -0,0 +1,226 @@
+// Copyright (c) 2023, QuantStack and Mamba Contributors
+//
+// Distributed under the terms of the BSD 3-Clause License.
+//
+// The full license is in the file LICENSE, distributed with this software.
+
+#ifndef MAMBA_SOLV_SOLVABLE_HPP
+#define MAMBA_SOLV_SOLVABLE_HPP
+
+#include <string>
+#include <string_view>
+#include <utility>
+
+#include <solv/pooltypes.h>
+
+#include "solv-cpp/ids.hpp"
+#include "solv-cpp/queue.hpp"
+
+typedef struct s_Solvable Solvable;
+
+namespace mamba::solv
+{
+    class ObjSolvableViewConst
+    {
+    public:
+
+        explicit ObjSolvableViewConst(const ::Solvable* solvable) noexcept;
+
+        auto raw() const -> const ::Solvable*;
+
+        auto id() const -> SolvableId;
+
+        auto name() const -> std::string_view;
+        auto version() const -> std::string_view;
+
+        auto build_number() const -> std::size_t;
+        auto build_string() const -> std::string_view;
+        auto file_name() const -> std::string_view;
+        auto license() const -> std::string_view;
+        auto md5() const -> std::string_view;
+        auto noarch() const -> std::string_view;
+        auto sha256() const -> std::string_view;
+        auto size() const -> std::size_t;
+        auto timestamp() const -> std::size_t;
+
+        /** Queue of ``DependencyId``. */
+        auto dependencies() const -> ObjQueue;
+        /** Queue of ``DependencyId``. */
+        auto provides() const -> ObjQueue;
+        /** Queue of ``DependencyId``. */
+        auto constraints() const -> ObjQueue;
+        /** Queue of ``StringId``. */
+        auto track_features() const -> ObjQueue;
+
+    private:
+
+        const ::Solvable* m_solvable = nullptr;
+    };
+
+    class ObjSolvableView : public ObjSolvableViewConst
+    {
+    public:
+
+        using raw_str_view = const char*;
+
+        explicit ObjSolvableView(::Solvable* repo) noexcept;
+
+        auto raw() const -> ::Solvable*;
+
+        void set_name(std::string_view str) const;
+        void set_version(std::string_view str) const;
+
+        /** The following attributes need a call to @ref ObjRepoView::internalize. */
+        void set_build_number(std::size_t n) const;
+        void set_build_string(std::string_view bld) const;
+        void set_file_name(raw_str_view fn) const;
+        void set_file_name(const std::string& fn) const;
```

Comment:
Why not use `std::string_view` for all the const setters?
Also, why are these setters const?

Reply:
Unfortunately we cannot here because we need a `\0` terminated string...